python/neuroglancer/pipeline/task_queue.py

```diff
@@ -80,12 +94,61 @@ class QueueEmpty(LookupError):
         def __init__(self):
             super(LookupError, self).__init__('Queue Empty')
 
-    def __init__(self, project=PROJECT_NAME, queue_name=QUEUE_NAME, local=True):
+    def __init__(self, n_threads=40, project=PROJECT_NAME, queue_name=QUEUE_NAME, local=True):
         self._project = 's~' + project # unsure why this is necessary
         self._queue_name = queue_name
 
-        self.api =  build('taskqueue', 'v1beta2', credentials=google_credentials).tasks()
+        self.api = googleapiclient.discovery.build('taskqueue', 'v1beta2', credentials=google_credentials).tasks()
 
+        self._queue = Queue.Queue(maxsize=0) # infinite size
+        self._threads = ()
+        self._terminate = threading.Event()
+
+        self._start_threads(n_threads)
+
+    @property
+    def pending(self):
+        return self._queue.qsize()
+
+    def _start_threads(self, n_threads):
+        self._terminate.set()
+        self._terminate = threading.Event()
+
+        threads = []
+
+        for _ in xrange(n_threads):
+            worker = threading.Thread(
+                target=self._consume_queue, 
+                args=(self._terminate,)
+            )
+            worker.daemon = True
+            worker.start()
+            threads.append(worker)
+
+        self._threads = tuple(threads)
+
+    def _kill_threads(self):
+        self._queue.join() # if no threads were set the queue is always empty
+        self._terminate.set()
+        self._threads = ()
+
+    def _consume_queue(self, terminate_evt):
+        api = googleapiclient.discovery.build('taskqueue', 'v1beta2', credentials=google_credentials).tasks()
```

Comment:
I changed self.api to self._api, but I can only use self._api in _consume_queue if we are sure it's thread safe. I'll add a docstring.